tests/e2e/specs/wcpay/shopper/shopper-checkout-purchase.spec.js

```diff
@@ -15,35 +15,68 @@ import {
 	confirmCardAuthentication,
 } from '../../../utils/payments';
 
-import { shopperWCP } from '../../../utils';
-
-describe( 'Successful purchase', () => {
-	beforeEach( async () => {
-		await setupProductCheckout(
-			config.get( 'addresses.customer.billing' )
-		);
-	} );
-
-	afterAll( async () => {
-		// Clear the cart at the end so it's ready for another test
-		await shopperWCP.emptyCart();
-	} );
-
-	it( 'using a basic card', async () => {
-		const card = config.get( 'cards.basic' );
-		await fillCardDetails( page, card );
-		await shopper.placeOrder();
-		await expect( page ).toMatch( 'Order received' );
-	} );
-
-	it( 'using a 3DS card', async () => {
-		const card = config.get( 'cards.3ds' );
-		await fillCardDetails( page, card );
-		await expect( page ).toClick( '#place_order' );
-		await confirmCardAuthentication( page );
-		await page.waitForNavigation( {
-			waitUntil: 'networkidle0',
+import { shopperWCP, merchantWCP } from '../../../utils';
+
+const cardTestingPreventionStates = [
+	{ cardTestingPreventionEnabled: false },
+	{ cardTestingPreventionEnabled: true },
+];
+
+describe.each( cardTestingPreventionStates )(
+	'Successful purchase, non-site builder theme',
+	( { cardTestingPreventionEnabled } ) => {
+		beforeAll( async () => {
+			if ( cardTestingPreventionEnabled ) {
+				await merchant.login();
+				await merchantWCP.enableCardTestingProtection();
+				await merchant.logout();
+			}
+		} );
+
+		beforeEach( async () => {
+			await setupProductCheckout(
```

Comment:
Can we add `shopperWCP.emptyCart()` on `beforeEach` to ensure the test environment is properly setup

Reply:
Added it in https://github.com/Automattic/woocommerce-payments/pull/8164/commits/a07d017aa9c3852e1b23538fece0319c4c24b285.